lightning/src/ln/channel.rs

```diff
@@ -1121,37 +1124,278 @@ impl_writeable_tlv_based!(PendingChannelMonitorUpdate, {
 	(0, update, required),
 });
 
-/// The `ChannelPhase` enum describes the current phase in life of a lightning channel with each of
+/// The `Channel` enum describes the current phase in life of a lightning channel with each of
 /// its variants containing an appropriate channel struct.
-pub(super) enum ChannelPhase<SP: Deref> where SP::Target: SignerProvider {
+pub(super) enum Channel<SP: Deref> where SP::Target: SignerProvider {
 	UnfundedOutboundV1(OutboundV1Channel<SP>),
 	UnfundedInboundV1(InboundV1Channel<SP>),
 	#[allow(dead_code)] // TODO(dual_funding): Remove once creating V2 channels is enabled.
 	UnfundedV2(PendingV2Channel<SP>),
-	Funded(Channel<SP>),
+	Funded(FundedChannel<SP>),
 }
 
-impl<'a, SP: Deref> ChannelPhase<SP> where
+impl<SP: Deref> Channel<SP> where
 	SP::Target: SignerProvider,
 	<SP::Target as SignerProvider>::EcdsaSigner: ChannelSigner,
 {
-	pub fn context(&'a self) -> &'a ChannelContext<SP> {
+	pub fn context(&self) -> &ChannelContext<SP> {
+		match self {
+			Channel::Funded(chan) => &chan.context,
+			Channel::UnfundedOutboundV1(chan) => &chan.context,
+			Channel::UnfundedInboundV1(chan) => &chan.context,
+			Channel::UnfundedV2(chan) => &chan.context,
+		}
+	}
+
+	pub fn context_mut(&mut self) -> &mut ChannelContext<SP> {
+		match self {
+			Channel::Funded(ref mut chan) => &mut chan.context,
+			Channel::UnfundedOutboundV1(ref mut chan) => &mut chan.context,
+			Channel::UnfundedInboundV1(ref mut chan) => &mut chan.context,
+			Channel::UnfundedV2(ref mut chan) => &mut chan.context,
+		}
+	}
+
+	pub fn unfunded_context_mut(&mut self) -> Option<&mut UnfundedChannelContext> {
+		match self {
+			Channel::Funded(_) => { debug_assert!(false); None },
+			Channel::UnfundedOutboundV1(chan) => Some(&mut chan.unfunded_context),
+			Channel::UnfundedInboundV1(chan) => Some(&mut chan.unfunded_context),
+			Channel::UnfundedV2(chan) => Some(&mut chan.unfunded_context),
+		}
+	}
+
+	pub fn is_funded(&self) -> bool {
+		self.as_funded().is_some()
+	}
```

Comment:
nit: for a minimal more efficiency, could be unrolled to:
```suggestion
	pub fn is_funded(&self) -> bool {
		matches!(self, Channel::Funded(_channel))
	}
```

Reply:
Done.